### PR TITLE
Backward compatibility for `python-jsonschema`

### PR DESCRIPTION
**Pull request by adwait-godbole**
@Julian, I haven't included any pre-release versions in the `matrix-versions.json`. That's fine right?

----
📚 Documentation preview 📚: https://bowtie-json-schema--1428.org.readthedocs.build/en/1428/

### CODE REVIEW

File: implementations/python-jsonschema/bowtie_jsonschema.py [hunk 1]
```diff
@@ -3,14 +3,46 @@
...
 import sys
 import traceback
 
 from jsonschema.validators import validator_for
-import referencing.jsonschema
+
+jsonschema_version = metadata.version("jsonschema")
+
+
+def compare_versions(v1: str, v2: str) -> int:
```
Comment (Julian) [inline implementations/python-jsonschema/bowtie_jsonschema.py:18]: Here jsonschema *does* use Python versioning (and semver) so we have a single known implementation and version scheme -- so here we definitely should use an existing parser (probably the one you had from the packaging repo).

**Comment by adwait-godbole** [comment]
Ok to merge now ? After merge you will have to trigger the `images.yml` for just `python-jsonschema` without specifying any version and it would build all of it and later the `versioned-report.yml` 😅

**Comment by Julian** [comment]
> I haven't included any pre-release versions in the matrix-versions.json. That's fine right?

Yep -- though I think you could have gone back further :) -- but we can always do that after.

**Comment by adwait-godbole** [comment]
> Yep -- though I think you could have gone back further :) -- but we can always do that after.

Further ? 

https://github.com/python-jsonschema/jsonschema/releases?page=7 is the last page right ? Here I only see upto `v4.0.0`

**Comment by Julian** [comment]
Seems like GitHub may not keep all releases or something, but here's the full release history: https://pypi.org/project/jsonschema/#history

**Comment by adwait-godbole** [comment]
Ohh wow. Okay will cover these versions later then. Thanks!

**Comment by adwait-godbole** [comment]
Just a reminder, can you also trigger the `verioned-report.yml`  please 😅?

**Comment by Julian** [comment]
Done!

**Comment by adwait-godbole** [comment]
Thanks!

**Pull request merged**